packages/client/src/components/SafeSettings/components/Assets/index.js

```diff
@@ -1,56 +1,173 @@
+  const onRemoveCollectionSubmit = async (identifier) => {
+    try {
+      await removeCollection(identifier);
+      closeModal();
+    } catch (error) {
+      showErrorModal(error);
+    }
   };
+
   const openAddVaultModal = () => {
     openModal(
-      <AddVault onCancel={() => closeModal()} onNext={onAddVaultSubmit} />
+      <AddVault
+        address={address}
+        onCancel={() => closeModal()}
+        onNext={onAddVaultSubmit}
+      />,
+      { headerTitle: "Add Token Vault" }
     );
   };
+
   const openAddCollectionModal = () => {
     openModal(
       <AddCollection
         onCancel={() => closeModal()}
         onNext={onAddCollectionSubmit}
-      />
+      />,
+      { headerTitle: "Add NFT Collection" }
     );
   };
+
+  const openRemoveVaultModal = (identifier) => {
+    const { displayName, tokenAddress } = getTokenMeta(identifier);
+    openModal(
+      <RemoveAsset
+        name={displayName}
+        address={formatAddress(tokenAddress)}
+        explanation={
+          "This is only possible if your balance for this token is 0."
+        }
+        onCancel={() => closeModal()}
+        onNext={() => onRemoveVaultSubmit(identifier)}
+      />,
+      { headerTitle: "Remove Token Vault" }
+    );
+  };
+
+  const openRemoveCollectionModal = (identifier) => {
+    const { contractName, contractAddress } = parseIdentifier(identifier);
+    openModal(
+      <RemoveAsset
+        name={contractName}
+        address={formatAddress(contractAddress)}
+        explanation={
+          "This is only possible if you don’t have any NFT from this collection in your treasury."
+        }
+        onCancel={() => closeModal()}
+        onNext={() => onRemoveCollectionSubmit(identifier)}
+      />,
+      { headerTitle: "Remove NFT Collection" }
+    );
+  };
+
   return (
-    <div className="p-0 mt-5">
-      <h4 className="is-size-5">Assets</h4>
-      <button
-        className="button mt-4 is-full is-secondary is-small with-icon is-align-self-flex-end"
-        onClick={openAddVaultModal}
-      >
-        Add Vault
-        <Plus />
-      </button>
-      <button
-        className="button mt-4 is-full is-secondary is-small with-icon is-align-self-flex-end"
-        onClick={openAddCollectionModal}
-      >
-        Add Collection
-        <Plus />
-      </button>
+    <div>
+      <div className="column p-0 mt-5 is-flex is-full is-justify-content-space-between">
+        <h4 className="is-size-5">Token Vaults</h4>
+        <button
+          className="button is-secondary is-small with-icon"
+          onClick={openAddVaultModal}
+        >
+          Add Token Vault
+          <Plus style={{ position: "relative", left: 5 }} />
+        </button>
+      </div>
+      <div className="column p-0 mt-4 is-flex is-flex-direction-column is-full rounded-sm border-light has-shadow">
+        <AssetTableView
+          assets={vaults && vaults[address] && Object.values(vaults[address])}
```

Comment:
I'm confused on this line, looking at AssetTableView, we are expecting a array, this looks like a boolean to me

Reply:
Changed the condition to
`vaults[address] && Object.values(vaults[address])`
vaults[address] will be either undefined, empty array or array with some values -> and when it comes to rendering the AssetTableView it's useful to let the component handle if the vaults are empty or the rows should be rendered.